src/util.c

```diff
@@ -709,6 +714,86 @@ dup_over_stdio(int *fdp){
 	return ret;
 }
 
+/**
+ * Generated random bytes.
+ *
+ * \param size Number of bytes to be genrated .
+ *
+ * \return \c Newly allocated buffer on success, NULL otherwise.
+ **/
+uint8_t *
+get_random_bytes(uint num) {
+	int fd;
+	char *file = "/dev/urandom";
+	uint8_t *buf;
+
+	buf = g_malloc0(num);
+
+	fd = open(file, O_RDONLY);
+	if (fd < 0) {
+		g_critical("Error opening %s: %s", file, strerror(errno));
+		return NULL;
+	}
+
+	if (read(fd, buf, num) == -1) {
+		g_critical("Error reading %s: %s", file, strerror(errno));
+		close(fd);
+		return NULL;
+	}
+
+	close(fd);
+	return buf;
+}
+
+// Legacy naming scheme for virtio block devices
+// Reference : https://github.com/torvalds/linux/blob/master/drivers/block/virtio_blk.c#L478
```

Comment:
Ah - I see that you've copied this verbatim (with a very minor tweak to appease a compiler warning?) Don't we atleast need a copyright attribution here for the FSF?